pageserver/src/tenant_mgr.rs

```diff
@@ -141,30 +144,42 @@ pub fn init_tenant_mgr(
     tenants_state::set_timeline_update_sender(timeline_updates_sender)?;
     walreceiver::init_wal_receiver_main_thread(conf, timeline_updates_receiver)?;
 
-    let SyncStartupData {
-        remote_index,
-        local_timeline_init_statuses,
-    } = storage_sync::start_local_timeline_sync(conf, remote_storage)
-        .context("Failed to set up local files sync with external storage")?;
-
-    let mut tenants_to_attach: HashMap<ZTenantId, Vec<(ZTimelineId, TimelineMetadata)>> =
-        HashMap::with_capacity(local_timeline_init_statuses.len());
-    for (tenant_id, local_timeline_init_statuses) in local_timeline_init_statuses {
-        for (timeline_id, init_status) in local_timeline_init_statuses {
-            match init_status {
-                LocalTimelineInitStatus::LocallyComplete(metadata) => tenants_to_attach
-                    .entry(tenant_id)
-                    .or_default()
-                    .push((timeline_id, metadata)),
-                LocalTimelineInitStatus::NeedsSync => {
-                    debug!(
-                        "timeline {tenant_id} for tenant {timeline_id} needs sync, \
-                         so skipped for adding into repository until sync is finished"
-                    );
-                }
-            }
+    let local_tenant_files = local_tenant_timeline_files(conf)
+        .context("Failed to collect local tenant timeline files")?;
+
+    let (remote_index, tenants_to_attach) = match remote_storage
+        .zip(conf.remote_storage_config.as_ref())
+    {
```

Comment:
Took me a while to understand what this `zip` does here. It's a bit too clever, IMHO. I'd suggest just:
```
if let Some(storage) = remote_storage {
    let storage_config = remote_storage_config.expect("remote storage without config");
```
or the same with `ok_or` if you're allergic to panics that can't happen.